--- procinfo-17/procinfo.c.smpfix	Fri Aug 27 10:40:23 1999
+++ procinfo-17/procinfo.c	Fri Aug 27 10:42:23 1999
@@ -226,6 +226,14 @@
 	    new.cpu_nice = VAL;
 	    new.cpu_sys = VAL;
 	    new.cpu_idle = VAL;
+	    /*
+	     * according to bug #1959, sometimes the cpu_idle
+	     * seems to go backwards(!) on SMP boxes.  This may
+	     * have been fixed in later kernel releases.  This
+	     * is just a sanity check.
+	     */
+	    if (new.cpu_idle < old.cpu_idle)
+	      new.cpu_idle = old.cpu_idle;
 	} else if (ISSTR ("disk")) {
 	    new.disk[0] = VAL;
 	    new.disk[1] = VAL;
